### PR TITLE
DeployERC20 functionality for wallet and wallet_address 

### PR DESCRIPTION
**Pull request by rohan-agarwal-coinbase**
### What changed? Why?
- Updated the `smart_contract.ts` with the SmartContract model with the appropriate getters and the ability to sign/broadcast/wait on it
- Created new types in `types.ts` to wrap the underlying SmartContractType` and `SmartContractOptions`, and to define the `SmartContractAPIClient` functions
- Renamed the existing `smartContract` API client to `contractEvent` in `coinbase.ts`
- Added new `smartContract` API Client which is constructed from `SmartContractsApiFactory`
- Updated `wallet_address.ts` with the new `deployToken` function
- Updated `wallet.ts` with the new `deployToken` function'
- Updated tests, pretty much copying the same pattern as `contract_invocation` tests

### Testing
`npm run test` succeeds. I also deployed a new token through the local `cdp-service`

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: src/coinbase/smart_contract.ts [hunk 1]
```diff
@@ -1,10 +1,220 @@
...
+    return this.model.options as NFTContractOptions;
+  }
+
+  /**
+   * Returns the ABI of the smart contract.
+   *
+   * @returns The ABI as a JSON-encoded string.
+   */
+  public getAbi(): string {
+    return this.model.abi;
```
Comment (alex-stone) [inline src/coinbase/smart_contract.ts:179]: Do we want to `JSON.parse` this object and return the object?

https://github.com/coinbase/coinbase-sdk-nodejs/blob/44abe9858c58664c799dade7bbf75bb23f14a1a6/src/coinbase/contract_invocation.ts#L110-L114

File: src/coinbase/smart_contract.ts [hunk 2]
```diff
@@ -1,10 +1,220 @@
...
+  }
+
+  /**
+   * Broadcasts the SmartContract deployment to the Network.
+   *
+   * @returns The SmartContract object
+   * @throws {APIError} if the API request to broadcast a SmartContract deployment fails.
+   */
+  public async broadcast(): Promise<SmartContract> {
+    if (!this.getTransaction()?.isSigned())
```
Comment (alex-stone) [inline src/coinbase/smart_contract.ts:209]: Is the `?` necessary if the typing of `getTransaction` is not nullable?

File: src/coinbase/smart_contract.ts (hunk 1 above)
Comment (rohan-agarwal-coinbase) [reply src/coinbase/smart_contract.ts]: good catch, fixed in updated PR!

File: src/coinbase/smart_contract.ts (hunk 2 above)
Comment (rohan-agarwal-coinbase) [reply src/coinbase/smart_contract.ts]: fixed!

**Review by alex-stone — APPROVED** [review]

**Pull request merged**